src/bridge/Makefile.am

```diff
@@ -3,10 +3,11 @@
 
 noinst_LIBRARIES += libcockpit-metrics.a
 
-libcockpit_metrics_a_CFLAGS = \
+libcockpit_metrics_a_CPPFLAGS = \
 	-DG_LOG_DOMAIN=\"cockpit-bridge\" \
-	$(COCKPIT_BRIDGE_CFLAGS) \
-	$(NULL)
+	$(glib_CFLAGS) \
+	$(json_glib_CFLAGS) \
+	$(AM_CPPFLAGS)
```

Comment:
This is more philosophical, but does it make sense to include `*_CFLAGS` into `CPPFLAGS`? Also, do we need to give `AM_CFLAGS` explicitly? If not, why is that implied, but `AM_CPPFLAGS` is not? Should this be split into two _CPPFLAGS and _CFLAGS rules perhaps?

It's the same for other paragraphs below, but let's discuss it here first.

Reply:
The reason for this is that the `--cflags` parameter to `pkg-config` isn't as specific about the distinction between `CFLAGS` and `CPPFLAGS` as automake is.  As a result, most-to-all of what ends up in `glib_CFLAGS` is actually flags for the preprocessor.  That's mostly `-I` paths, but also `-pthread` which is also a flag for the preprocessor (which results in definition of some additional macros).

At the end of the day it's pretty academic: no part of our build process calls the preprocessor separately from the comiler and these two things always end up being pasted together.  In my mind though, I'd prefer to reserve the `CFLAGS` for things like debugging, optimisation, and warnings flags (which are common to all of our targets).

Reply:
I just realised though that the way I define the `CFLAGS` for `cockpit-tls` is technically incorrect as per the above definitions.  I'm going to rework that.  I think I actually want to add `-pthread` to our top-level project-wide `CFLAGS` (yes, I know I just said that it's actually a preprocessor flag, but .. eh... it's a weird case).  It doesn't actually do anything anymore as far as I can tell (since all code is reentrant by default now) but, strictly speaking, it's correct.

Reply:
Thanks, pkg-config not separating these makes sense.

There's `AM_CPPFLAGS` for project-wide flags already anyway, so `-pthread` might be appropriate there?

So that leaves my question about why `$(AM_CPPFLAGS)` needs to be specified explicitly, but `$(AM_CFLAGS)` doesn't?

Reply:
the rule is that if you specify a per-target `foo_CFLAGS` then it uses that, otherwise it uses `AM_CFLAGS`.  ditto `foo_CPPFLAGS` and `AM_CPPFLAGS`.  That's why I'm always careful to include `AM_CPPFLAGS` in the `foo_CPPFLAGS` overrides.  Since we generally *don't* override `foo_CFLAGS` we get the default `AM_CFLAGS` automatically.

That said, I just found a few places where that isn't true (specifically, when we add `-fPIC`).  I'll fix those up in the next round :)